### PR TITLE
chore: add markdown link checker

### PR DESCRIPTION
**Pull request by miparnisari**
## Description

Add markdown link checker to ensure our readme doesn't have broken links

## References
n/a

## Review Checklist
- [x] I have added documentation for new/changed functionality in this PR or in [openfga.dev](https://github.com/openfga/openfga.dev)
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -5,6 +5,14 @@ on:
       - main
 
 jobs:
+  markdown-link-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
```
Comment (elbuo8) [inline .github/workflows/pull_request.yaml:11]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by craigpastro — APPROVED** [review]
LGTM

**Comment by jon-whit** [comment]
Why not introduce [yamllint](https://yamllint.readthedocs.io/en/stable/) for this instead? Then we can have yaml linting benefits outside of just the scope of Github actions.

I vote we introduce `.yamllint`.

**Comment by miparnisari** [comment]
> Why not introduce [yamllint](https://yamllint.readthedocs.io/en/stable/) for this instead? Then we can have yaml linting benefits outside of just the scope of Github actions.
> 
> Then we could use this action instead: https://github.com/marketplace/actions/yamllint-github-action
> 
> I vote we introduce `.yamllint`.

i'm confused, why do i need yaml linting?

**Comment by jon-whit** [comment]
> > Why not introduce [yamllint](https://yamllint.readthedocs.io/en/stable/) for this instead? Then we can have yaml linting benefits outside of just the scope of Github actions.
> > Then we could use this action instead: https://github.com/marketplace/actions/yamllint-github-action
> > I vote we introduce `.yamllint`.
> 
> i'm confused, why do i need yaml linting?

Oh, I'm sorry I messed that up. This is for markdown. Same comment though. We should probably have a `.yamllint` (for yaml) and a `.markdownlint.yaml` (for markdown).  This way our IDEs and tooling can enforce lint rules for YAML and Markdown beyond the scope of Github actions.

**Comment by miparnisari** [comment]
> > > Why not introduce [yamllint](https://yamllint.readthedocs.io/en/stable/) for this instead? Then we can have yaml linting benefits outside of just the scope of Github actions.
> > > Then we could use this action instead: https://github.com/marketplace/actions/yamllint-github-action
> > > I vote we introduce `.yamllint`.
> > 
> > 
> > i'm confused, why do i need yaml linting?
> 
> Oh, I'm sorry I messed that up. This is for markdown. Same comment though. We should probably have a `.yamllint` (for yaml) and a `.markdownlint.yaml` (for markdown). This way our IDEs and tooling can enforce lint rules for YAML and Markdown beyond the scope of Github actions.

Sounds good, separate PR though

**Pull request merged**